src/org/loklak/client/APIGenerator.java

```diff
@@ -0,0 +1,84 @@
+package org.loklak.client;
+
+import org.json.JSONObject;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Proxy;
+
+import static org.loklak.tools.JsonIO.loadJson;
+import static org.loklak.tools.JsonIO.pushJson;
+
+public class APIGenerator {
+
+    // to restrict instantiation of class
+    private APIGenerator() {
+    }
+
+    public static <T> T createApiMethods(Class<T> service, final String baseUrl) {
+        return (T) Proxy.newProxyInstance(service.getClassLoader(), new Class<?>[]{service},
+                new InvocationHandler() {
```

Comment:
It would be better to create a separate (inner) class for the `InvocationHandler` instance type as this looks difficult to maintain.